src/main/java/org/spongepowered/api/command/spec/CommandSpec.java

```diff
@@ -24,439 +24,202 @@
-         * @return this
+         * @param executor The {@link CommandExecutor} that will run the command
+         * @return This builder, for chaining
          */
-        public Builder extendedDescription(@Nullable Text extendedDescription) {
-            this.extendedDescription = extendedDescription;
-            return this;
-        }
+        Builder executor(CommandExecutor executor);
 
         /**
-         * Sets the argument specification for this command. Generally, for a
-         * multi-argument command the {@link GenericArguments#seq(CommandElement...)}
-         * method is used to parse a sequence of args.
+         * Provides the description for this command.
+         *
+         * <p>A one line summary should be provided to
+         * {@link #description(Text)}</p>
          *
-         * @see GenericArguments
-         * @param args The arguments object to use
-         * @return this
+         * @param extendedDescription The description to use, or {@code null}
+         *                            for no description.
+         * @return This builder, for chaining
          */
-        public Builder arguments(CommandElement args) {
-            checkNotNull(args, "args");
-            this.args = args;
-            return this;
-        }
+        Builder extendedDescription(@Nullable Text extendedDescription);
 
         /**
-         * Sets the argument specification for this command. This method accepts
-         * a sequence of arguments. This is equivalent to calling {@code
-         * arguments(seq(args))}.
+         * The flags that this command should accept. See {@link Flags}.
          *
-         * @see GenericArguments
-         * @param args The arguments object to use
-         * @return this
+         * @param flags The {@link Flags} to accept
+         * @return This builder, for chaining
          */
-        public Builder arguments(CommandElement... args) {
-            checkNotNull(args, "args");
-            this.args = GenericArguments.seq(args);
-            return this;
-        }
+        Builder flags(Flags flags);
 
         /**
-         * Sets the input tokenizer to be used to convert input from a string
-         * into a list of argument tokens.
+         * Determines how an argument string should be split.
          *
-         * @see InputTokenizer for common input parser implementations
-         * @param parser The parser to use
-         * @return this
+         * <p>Defaults to splitting on spaces, ignoring spaces in quoted
+         * regions.</p>
+         *
+         * @param tokenizer The {@link InputTokenizer} to use
+         * @return This builder, for chaining
          */
-        public Builder inputTokenizer(InputTokenizer parser) {
-            checkNotNull(parser, "parser");
-            this.argumentParser = parser;
-            return this;
-        }
+        Builder inputTokenizer(InputTokenizer tokenizer);
```

Comment:
I haven’t seen a plugin use a style other than the default, I don’t think it’s well known about. There isn’t actually much of a difference between them anyway, as they are all space separated for the most part - it’s whether quotes play a part or not - at least with the default.

I’m wary about making help too cluttered for a single command. I think that this may be the responsibility of the plugin implementing the command instead if they deviate from the norm.